src/hooks/useGuildEvents.ts

```diff
@@ -0,0 +1,99 @@
+import useSWRImmutable from "swr/immutable"
+import { EventSourcesKey, OneOf, supportedEventSources } from "types"
+
+type GuildEvent = {
+  title: string
+  description: string
+  start: number
+  end: number
+  memberCount: number
+  eventType: EventSourcesKey
+  location: string
+  url: string
+  image: string
+}
+
+type DiscordEvent = {
+  id: string
+  guildId: string
+  channelId?: string
+  creatorId?: string
+  name: string
+  description?: string
+  scheduledStartTimestamp: number
+  scheduledEndTimestamp?: number
+  privacyLevel: number
+  status: number
+  entityType: number
+  entityId?: string
+  userCount?: number
+  creator?: number
+  entityMetadata?: {
+    location: string
+  }
+  image?: string
+}
+
+const DiscordEventToGuildEvent = (
```

Comment:
let's use camelCase for function names (`discordEventToGuildEvent`), and I think we could rename the `guildId` param to `serverId`